### PR TITLE
feat: add conversion from text to list

### PR DESCRIPTION
**Pull request by evanjmg**
Allows a user to convert a paragraph or heading to a list.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -776,4 +776,30 @@ export default class NestedList {
      */
     this.unshiftItem();
   }
+
+  static joinRecursive(data) {
```
Comment (neSpecc) [inline src/index.js:780]: please write a jsdoc

**Review by neSpecc — CHANGES_REQUESTED** [review]
I found that there are some excess dots: 

```
- One
- Two
```

Will become

```
One. . Two.
```



Also, please update a patch version.

**Comment by evanjmg** [comment]
Question - @editorjs/list adds periods. Should we even bother adding periods? What if you have a list with periods? Does it make sense to change the content?

File: src/index.js (hunk 1 above)
Comment (evanjmg) [reply src/index.js:785]: updated

**Comment by neSpecc** [comment]
> Question - @editorjs/list adds periods. Should we even bother adding periods? What if you have a list with periods? Does it make sense to change the content?

Good question. I think we need to start with something and collect feedback from users.

**Comment by evanjmg** [comment]
@neSpecc  any other changes needed or can this be merged?

**Comment by dfrankow** [comment]
> @neSpecc any other changes needed or can this be merged?

I have the same question.

**Review by talyguryn — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

**Pull request merged**